src/audio/google/google_rtc_audio_processing.c

```diff
@@ -602,18 +604,16 @@ static int google_rtc_audio_processing_prepare(struct processing_module *mod,
 		return -EINVAL;
 	}
 
+	struct comp_buffer *b0 = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
```

Comment:
So... I get that.  It's relying on a stale/deprecated API scheme.  But see the commit message.  The problem is that the mechanism currently being used to answer the question "Is this source/sink connected to a component on the same pipeline or a different one?" is just incorrect as it stands.  Those numbers are ephemeral values assigned at runtime in the kernel (e.g. they're like file descriptors), and the only reason they work here is because they've been hand-matched to the values chosen for this specific device with this topology.

So, I guess I'd argue we should pick the correct but to-be-replaced solution over the incorrect one.  For sure a proper API is required.  Basically this is a variant of a common SOF problem: the topology as it exists in the kernel is only incompletely exposed inside the firmware (everything gets cooked by the kernel, which drops a lot of information), so everywhere we're having to use trickery to recover it.